File: 002/e002.swift
// By considering the terms in the Fibonacci sequence whose values do not exceed four million,
// find the sum of the even-valued terms.
func fib(_ n : Int) -> Int {
    var a = 1
    var b = 1
    guard n >= 2 else {
        return a
    }
    for _ in 2...n {
        let tmp = a
        a = a + b
        b = tmp
    }
    return a
}

var n = 0
var term = 0
var sum = 0
while term < Int(4e6) {
    if term % 2 == 0 {
        sum += term
    }
    n += 1
    term = fib(n)
    print("n = \(n), fib(n) = \(term)")
}
print(sum)

File: 002/e002.go
package main

import "fmt"

// By considering the terms in the Fibonacci sequence whose values do not exceed four million,
// find the sum of the even-valued terms.
func fib(n int) int {
    var a = 1
    var b = 1
    if n < 2 {
        return a
    }
    for i := 2; i < n; i++ {
        tmp := a
        a = a + b
        b = tmp
    }
    return a
}

func main() {
    n := 0
    term := 0
    sum := 0
    for term < 4e6 {
        if term % 2 == 0 {
            sum += term
        }
        n += 1
        term = fib(n)
        fmt.Printf("n = %v, fib(n) = %v\n", n, term)
    }
    fmt.Println(sum)
}
